app/services/foreman_discovery/node_api/power.rb

```diff
@@ -0,0 +1,56 @@
+module ForemanDiscovery::NodeAPI
+  class Power
+    class << self
+      def service(data)
+        PowerService.new(data)
+      end
+
+      def legacy_direct_service(data)
+        PowerLegacyDirectService.new(data)
+      end
+
+      def legacy_proxied_service(data)
+        PowerLegacyProxiedService.new(data)
+      end
+    end
+  end
+
+  class PowerService < NodeResource
+    def url
+      @args[:url] + "/power"
+    end
+
+    def reboot
```

Comment:
Nah its a complete different operation with its own input. And also, this gave us chance to get rid of reusing BMC interface once and forever.